akka-http-core/src/main/scala/akka/http/impl/engine/client/PoolGateway.scala

```diff
@@ -27,6 +27,13 @@ private[http] final class PoolGateway(gatewayRef: ActorRef, val hcps: HostConnec
 
   private val shutdownCompletedPromise: Promise[Done] = Promise[Done]()
 
+  {
+    implicit val ec = fm.executionContext
+    shutdownCompletedPromise.future.foreach { _ ⇒
+      gatewayRef ! Shutdown(this, shutdownCompletedPromise)
+    }
+  }
```

Comment:
Good, we could actually probably execute it on the same thread, but I don't think performance matters here so this is fine.

Reply:
OK, will do